src/overlays/actors/ovl_Obj_Driftice/z_obj_driftice.c

```diff
@@ -76,200 +87,257 @@ static InitChainEntry sInitChain[] = {
     ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_STOP),
 };
 
-void func_80A66570(ObjDriftice* this, s32 arg1) {
-    Math_Vec3s_ToVec3f(&this->dyna.actor.world.pos, &this->pathPoints[arg1]);
+void ObjDriftice_SetPosToPathPoint(ObjDriftice* this, s32 pathPointIdx) {
+    Math_Vec3s_ToVec3f(&this->dyna.actor.world.pos, &this->pathPoints[pathPointIdx]);
 }
 
-void func_80A665AC(s16* arg0, s16 arg1) {
-    if (arg1 < *arg0) {
-        *arg0 = arg1;
-    } else if (*arg0 < -arg1) {
-        *arg0 = -arg1;
+void ObjDriftice_ClampSymetricAroundZero(s16* rot, s16 threshold) {
+    if (*rot > threshold) {
+        *rot = threshold;
+    } else if (*rot < -threshold) {
+        *rot = -threshold;
     }
 }
 
-f32 func_80A665EC(ObjDrifticeStruct* arg0, ObjDriftice* this) {
-    f32 phi_f2 = 0.0f;
-    f32 sp20;
-
-    if (this->unk_248 >= 10) {
-        sp20 = -8.0f;
-    } else if (this->unk_248 < -9) {
-        sp20 = 0.0f;
-    } else if (this->unk_248 > 0) {
-        sp20 = (Math_CosS(fabsf(this->unk_248) * (0x8000 * 0.1f)) - 1.0f) * 4.0f;
+/*
+Returns the y offset contribution related to player interactions.
+
+Updates an oscillation state and uses output. Combines with another term based on how long player has been stood
+ on platform for.
+*/
+f32 ObjDriftice_PlayerRelatedYOffsetPhysicsUpdate(ObjDrifticeOscillationState* oscillationState, ObjDriftice* this) {
+    f32 zeroF = 0.0f;
+    f32 yOffset;
+
+    if (this->timerPlayerOnTop >= Y_OFFSET_CHANGE_TIME) {
+        yOffset = -8.0f;
+    } else if (this->timerPlayerOnTop <= -Y_OFFSET_CHANGE_TIME) {
+        yOffset = 0.0f;
+    } else if (this->timerPlayerOnTop > 0) {
+        yOffset = (Math_CosS(fabsf(this->timerPlayerOnTop) * (0x8000 * (1.0f / Y_OFFSET_CHANGE_TIME))) - 1.0f) *
+                  4.0f; // Decrease from 0 to -8
```

Comment:
I think it's in the spirit of what engineer said, but try to put comments above code and not on the same line. Just wanted to show you an example where it occured.